Can't move a Dataverse containing datasets

```
Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Long
	at java.util.Vector.forEach(Vector.java:1275)
	at edu.harvard.iq.dataverse.engine.command.impl.MoveDataverseCommand.executeImpl(MoveDataverseCommand.java:95)
	at edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand.execute(AbstractVoidCommand.java:29)
	at edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand.execute(AbstractVoidCommand.java:1)
	at edu.harvard.iq.dataverse.EjbDataverseEngine.submit(EjbDataverseEngine.java:251)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```

--- a/dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/Dataset.java
+++ b/dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/Dataset.java
@@ -53,10 +53,6 @@ import java.util.stream.Stream;
                 query = "SELECT d FROM Dataset d WHERE d.identifier=:identifier"),
         @NamedQuery(name = "Dataset.findByIdentifierAuthorityProtocol",
                 query = "SELECT d FROM Dataset d WHERE d.identifier=:identifier AND d.protocol=:protocol AND d.authority=:authority"),
-        @NamedQuery(name = "Dataset.findIdByOwnerId",
-                query = "SELECT o.identifier FROM Dataset o WHERE o.owner.id=:ownerId"),
-        @NamedQuery(name = "Dataset.findByOwnerId",
-                query = "SELECT o FROM Dataset o WHERE o.owner.id=:ownerId"),
 })
 
 /*

--- a/dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetRepository.java
+++ b/dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetRepository.java
@@ -4,6 +4,8 @@ import edu.harvard.iq.dataverse.persistence.JpaRepository;
 
 import javax.ejb.Singleton;
 
+import java.util.List;
+
 @Singleton
 public class DatasetRepository extends JpaRepository<Long, Dataset> {
 
@@ -12,4 +14,19 @@ public class DatasetRepository extends JpaRepository<Long, Dataset> {
     public DatasetRepository() {
         super(Dataset.class);
     }
+
+    // -------------------- LOGIC --------------------
+
+    public List<Dataset> findByOwnerId(Long ownerId) {
+        return em.createQuery("SELECT o FROM Dataset o WHERE o.owner.id=:ownerId", Dataset.class)
+                .setParameter("ownerId", ownerId)
+                .getResultList();
+    }
+
+    public List<Long> findIdsByOwnerId(Long ownerId) {
+        return em.createQuery("SELECT o.id FROM Dataset o WHERE o.owner.id=:ownerId", Long.class)
+                .setParameter("ownerId", ownerId)
+                .getResultList();
+    }
+
 }

--- a/dataverse-persistence/src/main/resources/Bundle_pl.properties
+++ b/dataverse-persistence/src/main/resources/Bundle_pl.properties
@@ -2839,7 +2839,7 @@ dahsboard.datamove.dataverse.command.error.dataverse.template.not.in.target.data
 dahsboard.datamove.dataverse.command.error.dataverse.featured.in.current.dataverse=Kolekcja, kt\u00F3r\u0105 pr\u00F3bujesz przenie\u015B\u0107 jest wyr\u00F3\u017Cniona w swojej aktualnej kolekcji nadrz\u0119dnej.
 dahsboard.datamove.dataverse.command.error.dataverse.metadata.block.not.in.target.dataverse=Blok metadanych w kolekcji, kt\u00F3r\u0105 pr\u00F3bujesz przenie\u015B\u0107 nie jest obecny w nowej kolekcji nadrz\u0119dnej.
 dahsboard.datamove.dataverse.command.error.dataverse.linked.to.target.dataverse=Kolekcja, kt\u00F3r\u0105 pr\u00F3bujesz przenie\u015B\u0107 jest przypi\u0119ta do nowej kolekcji nadrz\u0119dnej lub kt\u00F3rej\u015B z jej kolekcji wy\u017Cszego rz\u0119du.
-dahsboard.datamove.dataverse.command.error.dataset.linked.to.target.dataverse=Zbi\u00F3r danych w kolekcji, kt\u00F3r\u0105 pr\u00F3bujesz przenie\u015B\u0107 jest przypi\u0119ty do nowej kolekcji narz\u0119dnej lub kt\u00F3rej\u015B z jej jej kolekcji wy\u017Cszego rz\u0119du.
+dahsboard.datamove.dataverse.command.error.dataset.linked.to.target.dataverse=Zbi\u00F3r danych w kolekcji, kt\u00F3r\u0105 pr\u00F3bujesz przenie\u015B\u0107 jest przypi\u0119ty do nowej kolekcji narz\u0119dnej lub kt\u00F3rej\u015B z jej kolekcji wy\u017Cszego rz\u0119du.
 dashboard.datamove.dataverse.command.suggestForce=Mo\u017Cesz wymusi\u0107 to polecenie zaznaczaj\u0105c opcj\u0119 "Wymu\u015B przeniesienie".
 dashboard.datamove.dataverse.message.success=Kolekcja "{0}" zosta\u0142a pomy\u015Blnie przeniesiona do kolekcji "{1}".
 dashboard.datamove.dataverse.message.failure.summary=Przeniesienie kolekcji nie powiod\u0142o si\u0119.

--- a/dataverse-persistence/src/test/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetRepositoryIT.java
+++ b/dataverse-persistence/src/test/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetRepositoryIT.java
@@ -1,0 +1,80 @@
+package edu.harvard.iq.dataverse.persistence.dataset;
+
+import edu.harvard.iq.dataverse.persistence.PersistenceArquillianDeployment;
+import org.junit.Test;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.hamcrest.Matchers.*;
+
+public class DatasetRepositoryIT extends PersistenceArquillianDeployment {
+
+    @Inject
+    private DatasetRepository datasetRepository;
+
+
+    //-------------------- TESTS --------------------
+
+    @Test
+    public void findByOwnerId() {
+        // when
+        List<Dataset> datasets = datasetRepository.findByOwnerId(19L);
+        // then
+        assertNotNull(datasets);
+        assertEquals(2, datasets.size());
+        assertThat(datasets.stream().map(Dataset::getId).collect(toList()), containsInAnyOrder(56L, 57L));
+    }
+
+    @Test
+    public void findByOwnerId_dataverse_without_datasets() {
+        // when
+        List<Dataset> datasets = datasetRepository.findByOwnerId(67L);
+        // then
+        assertNotNull(datasets);
+        assertEquals(0, datasets.size());
+    }
+
+    @Test
+    public void findByOwnerId_non_existing_dataverse() {
+        // when
+        List<Dataset> datasets = datasetRepository.findByOwnerId(9999L);
+        // then
+        assertNotNull(datasets);
+        assertEquals(0, datasets.size());
+    }
+
+
+    @Test
+    public void findIdsByOwnerId() {
+        // when
+        List<Long> datasetIds = datasetRepository.findIdsByOwnerId(19L);
+        // then
+        assertNotNull(datasetIds);
+        assertEquals(2, datasetIds.size());
+        assertThat(datasetIds, containsInAnyOrder(56L, 57L));
+    }
+
+    @Test
+    public void findIdsByOwnerId_dataverse_without_datasets() {
+        // when
+        List<Long> datasetIds = datasetRepository.findIdsByOwnerId(67L);
+        // then
+        assertNotNull(datasetIds);
+        assertEquals(0, datasetIds.size());
+    }
+
+    @Test
+    public void findIdsByOwnerId_non_existing_dataverse() {
+        // when
+        List<Long> datasetIds = datasetRepository.findIdsByOwnerId(9999L);
+        // then
+        assertNotNull(datasetIds);
+        assertEquals(0, datasetIds.size());
+    }
+}

--- a/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DatasetDao.java
+++ b/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DatasetDao.java
@@ -11,6 +11,7 @@ import edu.harvard.iq.dataverse.persistence.datafile.DataFile;
 import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
 import edu.harvard.iq.dataverse.persistence.dataset.DatasetField;
 import edu.harvard.iq.dataverse.persistence.dataset.DatasetLock;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetRepository;
 import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
 import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersionUser;
 import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
@@ -30,13 +31,11 @@ import javax.ejb.TransactionAttributeType;
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.persistence.EntityManager;
-import javax.persistence.NoResultException;
 import javax.persistence.PersistenceContext;
 import javax.persistence.Query;
 import javax.persistence.StoredProcedureQuery;
 import javax.persistence.TypedQuery;
 import java.io.InputStream;
-import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
@@ -73,7 +72,8 @@ public class DatasetDao implements java.io.Serializable {
     @EJB
     EjbDataverseEngine commandEngine;
 
-    private static final SimpleDateFormat logFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH-mm-ss");
+    @EJB
+    private DatasetRepository datasetRepository;
 
     @PersistenceContext(unitName = "VDCNet-ejbPU")
     protected EntityManager em;
@@ -83,49 +83,7 @@ public class DatasetDao implements java.io.Serializable {
     }
 
     public List<Dataset> findByOwnerId(Long ownerId) {
-        return findByOwnerId(ownerId, false);
-    }
-
-    public List<Dataset> findPublishedByOwnerId(Long ownerId) {
-        return findByOwnerId(ownerId, true);
-    }
-
-    private List<Dataset> findByOwnerId(Long ownerId, boolean onlyPublished) {
-        List<Dataset> retList = new ArrayList<>();
-        TypedQuery<Dataset> query = em.createNamedQuery("Dataset.findByOwnerId", Dataset.class);
-        query.setParameter("ownerId", ownerId);
-        if (!onlyPublished) {
-            return query.getResultList();
-        } else {
-            for (Dataset ds : query.getResultList()) {
-                if (ds.isReleased() && !ds.isDeaccessioned()) {
-                    retList.add(ds);
-                }
-            }
-            return retList;
-        }
-    }
-
-    public List<Long> findIdsByOwnerId(Long ownerId) {
-        return findIdsByOwnerId(ownerId, false);
-    }
-
-    private List<Long> findIdsByOwnerId(Long ownerId, boolean onlyPublished) {
-        List<Long> retList = new ArrayList<>();
-        if (!onlyPublished) {
-            return em.createNamedQuery("Dataset.findIdByOwnerId")
-                    .setParameter("ownerId", ownerId)
-                    .getResultList();
-        } else {
-            List<Dataset> results = em.createNamedQuery("Dataset.findByOwnerId")
-                    .setParameter("ownerId", ownerId).getResultList();
-            for (Dataset ds : results) {
-                if (ds.isReleased() && !ds.isDeaccessioned()) {
-                    retList.add(ds.getId());
-                }
-            }
-            return retList;
-        }
+        return datasetRepository.findByOwnerId(ownerId);
     }
 
     public List<Dataset> findAll() {
@@ -267,35 +225,6 @@ public class DatasetDao implements java.io.Serializable {
                 .setParameter("authority", dataset.getAuthority())
                 .setParameter("protocol", dataset.getProtocol())
                 .getResultList().isEmpty();
-    }
-
-    public Long getMaximumExistingDatafileIdentifier(Dataset dataset) {
-        //Cannot rely on the largest table id having the greatest identifier counter
-        long zeroFiles = new Long(0);
-        Long retVal = zeroFiles;
-        Long testVal;
-        List<Object> idResults;
-        Long dsId = dataset.getId();
-        if (dsId != null) {
-            try {
-                idResults = em.createNamedQuery("Dataset.findIdByOwnerId")
-                        .setParameter("ownerId", dsId).getResultList();
-            } catch (NoResultException ex) {
-                logger.log(Level.FINE, "No files found in dataset id {0}. Returning a count of zero.", dsId);
-                return zeroFiles;
-            }
-            if (idResults != null) {
-                for (Object raw : idResults) {
-                    String identifier = (String) raw;
-                    identifier = identifier.substring(identifier.lastIndexOf("/") + 1);
-                    testVal = new Long(identifier);
-                    if (testVal > retVal) {
-                        retVal = testVal;
-                    }
-                }
-            }
-        }
-        return retVal;
     }
 
     public DatasetVersion storeVersion(DatasetVersion dsv) {
@@ -598,80 +527,6 @@ public class DatasetDao implements java.io.Serializable {
         commandEngine.submit(new FinalizeDatasetPublicationCommand(theDataset, request, isPidPrePublished));
     }
 
-    /*
-     Experimental asynchronous method for requesting persistent identifiers for 
-     datafiles. We decided not to run this method on upload/create (so files 
-     will not have persistent ids while in draft; when the draft is published, 
-     we will force obtaining persistent ids for all the files in the version. 
-     
-     If we go back to trying to register global ids on create, care will need to 
-     be taken to make sure the asynchronous changes below are not conflicting with 
-     the changes from file ingest (which may be happening in parallel, also 
-     asynchronously). We would also need to lock the dataset (similarly to how 
-     tabular ingest logs the dataset), to prevent the user from publishing the
-     version before all the identifiers get assigned - otherwise more conflicts 
-     are likely. (It sounds like it would make sense to treat these two tasks -
-     persistent identifiers for files and ingest - as one post-upload job, so that 
-     they can be run in sequence). -- L.A. Mar. 2018
-    */
-    @Asynchronous
-    public void obtainPersistentIdentifiersForDatafiles(Dataset dataset) {
-        GlobalIdServiceBean idServiceBean = GlobalIdServiceBean.getBean(dataset.getProtocol(), commandEngine.getContext());
-
-        //If the Id type is sequential and Dependent then write file idenitifiers outside the command
-        String datasetIdentifier = dataset.getIdentifier();
-        Long maxIdentifier = null;
-
-        if (isDataFilePIDSequentialDependent()) {
-            maxIdentifier = getMaximumExistingDatafileIdentifier(dataset);
-        }
-
-        for (DataFile datafile : dataset.getFiles()) {
-            logger.info("Obtaining persistent id for datafile id=" + datafile.getId());
-
-            if (datafile.getIdentifier() == null || datafile.getIdentifier().isEmpty()) {
-
-                logger.info("Obtaining persistent id for datafile id=" + datafile.getId());
-
-                if (maxIdentifier != null) {
-                    maxIdentifier++;
-                    datafile.setIdentifier(datasetIdentifier + "/" + maxIdentifier.toString());
-                } else {
-                    datafile.setIdentifier(fileService.generateDataFileIdentifier(datafile, idServiceBean));
-                }
-
-                if (datafile.getProtocol() == null) {
-                    datafile.setProtocol(settingsService.getValueForKey(SettingsServiceBean.Key.Protocol));
-                }
-                if (datafile.getAuthority() == null) {
-                    datafile.setAuthority(settingsService.getValueForKey(SettingsServiceBean.Key.Authority));
-                }
-
-                logger.info("identifier: " + datafile.getIdentifier());
-
-                String doiRetString;
-
-                try {
-                    logger.log(Level.FINE, "creating identifier");
-                    doiRetString = idServiceBean.createIdentifier(datafile);
-                } catch (Throwable e) {
-                    logger.log(Level.WARNING, "Exception while creating Identifier: " + e.getMessage(), e);
-                    doiRetString = "";
-                }
-
-                // Check return value to make sure registration succeeded
-                if (!idServiceBean.registerWhenPublished() && doiRetString.contains(datafile.getIdentifier())) {
-                    datafile.setIdentifierRegistered(true);
-                    datafile.setGlobalIdCreateTime(new Date());
-                }
-
-                DataFile merged = em.merge(datafile);
-                merged = null;
-            }
-
-        }
-    }
-
     /**
      * @return true if dataset is In Review locked state
      */
@@ -690,11 +545,5 @@ public class DatasetDao implements java.io.Serializable {
             }
         }
         return false;
-    }
-
-    private boolean isDataFilePIDSequentialDependent() {
-        String doiIdentifierType = settingsService.getValueForKey(SettingsServiceBean.Key.IdentifierGenerationStyle);
-        String doiDataFileFormat = settingsService.getValueForKey(SettingsServiceBean.Key.DataFilePIDFormat);
-        return doiIdentifierType.equals("sequentialNumber") && doiDataFileFormat.equals("DEPENDENT");
     }
 }

--- a/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseDao.java
+++ b/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseDao.java
@@ -10,6 +10,7 @@ import edu.harvard.iq.dataverse.authorization.groups.GroupServiceBean;
 import edu.harvard.iq.dataverse.dataaccess.ImageThumbConverter;
 import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
 import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetRepository;
 import edu.harvard.iq.dataverse.persistence.dataset.MetadataBlock;
 import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
 import edu.harvard.iq.dataverse.persistence.dataverse.DataverseTheme;
@@ -54,34 +55,31 @@ public class DataverseDao implements java.io.Serializable {
 
     private static final Logger logger = Logger.getLogger(DataverseDao.class.getCanonicalName());
     @EJB
-    IndexServiceBean indexService;
+    private IndexServiceBean indexService;
 
     @EJB
-    AuthenticationServiceBean authService;
+    private AuthenticationServiceBean authService;
 
     @EJB
-    DatasetDao datasetDao;
+    private DatasetRepository datasetRepository;
 
     @EJB
-    DataverseLinkingDao dataverseLinkingService;
+    private DataverseLinkingDao dataverseLinkingService;
 
     @EJB
-    DatasetLinkingServiceBean datasetLinkingService;
+    private DatasetLinkingServiceBean datasetLinkingService;
 
     @EJB
-    GroupServiceBean groupService;
+    private GroupServiceBean groupService;
 
     @EJB
-    DataverseRoleServiceBean rolesService;
+    private DataverseRoleServiceBean rolesService;
 
     @EJB
-    PermissionServiceBean permissionService;
+    private PermissionServiceBean permissionService;
 
     @EJB
-    SystemConfig systemConfig;
-
-    @Inject
-    DataverseSession session;
+    private SystemConfig systemConfig;
 
     @PersistenceContext(unitName = "VDCNet-ejbPU")
     private EntityManager em;
@@ -262,73 +260,6 @@ public class DataverseDao implements java.io.Serializable {
         }
         return null;
     }
-    
-    /*
-    public boolean isDataverseLogoThumbnailAvailable(Dataverse dataverse, User user) {    
-        if (dataverse == null) {
-            return false; 
-        }
-                
-        // First, check if the dataverse has a defined logo: 
-        
-        //if (dataverse.getDataverseTheme() != null && dataverse.getDataverseTheme().getLogo() != null && !dataverse.getDataverseTheme().getLogo().equals("")) {
-            File dataverseLogoFile = getLogo(dataverse);
-            if (dataverseLogoFile != null) {
-                String logoThumbNailPath = null;
-
-                if (dataverseLogoFile.exists()) {
-                    logoThumbNailPath = ImageThumbConverter.generateImageThumbnailFromFile(dataverseLogoFile.getAbsolutePath(), 48);
-                    if (logoThumbNailPath != null) {
-                        return true;
-                    }
-                }
-            }
-        //}
-        */
-    // If there's no uploaded logo for this dataverse, go through its
-    // [released] datasets and see if any of them have card images:
-    //
-    // TODO:
-    // Discuss/Decide if we really want to do this - i.e., go through every
-    // file in every dataset below...
-    // -- L.A. 4.0 beta14
-        /*
-        for (Dataset dataset : datasetDao.findPublishedByOwnerId(dataverse.getId())) {
-            if (dataset != null) {
-                DatasetVersion releasedVersion = dataset.getReleasedVersion();
-                
-                if (releasedVersion != null) {
-                    if (datasetDao.isDatasetCardImageAvailable(releasedVersion, user)) {
-                        return true;
-                    }
-                }
-            }
-        }   */     
-        /*
-        return false; 
-    } */
-
-    private File getLogo(Dataverse dataverse) {
-        if (dataverse.getId() == null) {
-            return null;
-        }
-
-        DataverseTheme theme = dataverse.getDataverseTheme();
-        if (theme != null && theme.getLogo() != null && !theme.getLogo().isEmpty()) {
-            Properties p = System.getProperties();
-            String domainRoot = p.getProperty("com.sun.aas.instanceRoot");
-
-            if (domainRoot != null && !"".equals(domainRoot)) {
-                return new File(domainRoot + File.separator +
-                                        "docroot" + File.separator +
-                                        "logos" + File.separator +
-                                        dataverse.getLogoOwnerId() + File.separator +
-                                        theme.getLogo());
-            }
-        }
-
-        return null;
-    }
 
     private File getLogoById(Long id) {
         if (id == null) {
@@ -463,45 +394,6 @@ public class DataverseDao implements java.io.Serializable {
         return countQuery.getSingleResult();
     }
 
-    /**
-     * Used to identify and properly display Harvested objects on the dataverse page.
-     *//*
-    @Deprecated
-    public Map<Long, String> getAllHarvestedDataverseDescriptions(){
-        
-        String qstr = "SELECT dataverse_id, archiveDescription FROM harvestingClient;";
-        List<Object[]> searchResults = null;
-        
-        try {
-            searchResults = em.createNativeQuery(qstr).getResultList();
-        } catch (Exception ex) {
-            searchResults = null;
-        }
-        
-        if (searchResults == null) {
-            return null;
-        }
-        
-        Map<Long, String> ret = new HashMap<>();
-        
-        for (Object[] result : searchResults) {
-            Long dvId = null;
-            if (result[0] != null) {
-                try {
-                    dvId = (Long)result[0];
-                } catch (Exception ex) {
-                    dvId = null;
-                }
-                if (dvId == null) {
-                    continue;
-                }
-                
-                ret.put(dvId, (String)result[1]);
-            }
-        }
-        
-        return ret;        
-    }*/
     public String getParentAliasString(SolrSearchResult solrSearchResult) {
         Long dvId = solrSearchResult.getEntityId();
         String retVal = "";
@@ -603,16 +495,12 @@ public class DataverseDao implements java.io.Serializable {
         // get list of Dataverse children
         List<Long> dataverseChildren = findIdsByOwnerId(dvId);
         // get list of Dataset children
-        List<Long> datasetChildren = datasetDao.findIdsByOwnerId(dvId);
+        List<Long> datasetChildren = datasetRepository.findIdsByOwnerId(dvId);
 
-        if (dataverseChildren == null) {
-            return datasetChildren;
-        } else {
-            for (Long childDvId : dataverseChildren) {
-                datasetChildren.addAll(findAllDataverseDatasetChildren(childDvId));
-            }
-            return datasetChildren;
+        for (Long childDvId : dataverseChildren) {
+            datasetChildren.addAll(findAllDataverseDatasetChildren(childDvId));
         }
+        return datasetChildren;
     }
 
     public String addRoleAssignmentsToChildren(Dataverse owner, List<String> rolesToInherit,

--- a/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Access.java
+++ b/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Access.java
@@ -792,35 +792,6 @@ public class Access extends AbstractApiBean {
             }
         }
 
-        // If there's no uploaded logo for this dataverse, go through its 
-        // [released] datasets and see if any of them have card images:
-
-        // TODO: figure out if we want to be doing this! 
-        // (efficiency considerations...) -- L.A. 4.0 
-        // And we definitely don't want to be doing this for harvested 
-        // dataverses:
-        /*
-        StorageIO thumbnailDataAccess = null; 
-        
-        if (!dataverse.isHarvested()) {
-            for (Dataset dataset : datasetDao.findPublishedByOwnerId(dataverseId)) {
-                logger.info("dvCardImage: checking dataset "+dataset.getGlobalId());
-                if (dataset != null) {
-                    DatasetVersion releasedVersion = dataset.getReleasedVersion();
-                    logger.info("dvCardImage: obtained released version "+releasedVersion.getTitle());
-                    thumbnailDataAccess = getThumbnailForDatasetVersion(releasedVersion); 
-                    if (thumbnailDataAccess != null) {
-                        logger.info("dvCardImage: obtained thumbnail for the version.");
-                        break;
-                    }
-                }
-            }
-        }
-        
-        if (thumbnailDataAccess != null && thumbnailDataAccess.getInputStream() != null) {
-            return thumbnailDataAccess.getInputStream();
-        }
-        */
         return null;
     }
 

--- a/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java
+++ b/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java
@@ -76,6 +76,7 @@ import javax.validation.ConstraintViolation;
 import javax.validation.ConstraintViolationException;
 import javax.ws.rs.Consumes;
 import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.PUT;
@@ -998,7 +999,9 @@ public class Dataverses extends AbstractApiBean {
 
     @POST
     @Path("{id}/move/{targetDataverseAlias}")
-    public Response moveDataverse(@PathParam("id") String id, @PathParam("targetDataverseAlias") String targetDataverseAlias, @QueryParam("forceMove") Boolean force) {
+    public Response moveDataverse(@PathParam("id") String id,
+                    @PathParam("targetDataverseAlias") String targetDataverseAlias, 
+                    @QueryParam("forceMove") @DefaultValue("false") boolean force) {
         try {
             User u = findUserOrDie();
             Dataverse dv = findDataverseOrDie(id);

--- a/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/MoveDataverseCommand.java
+++ b/dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/MoveDataverseCommand.java
@@ -46,9 +46,9 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
     private static final Logger logger = Logger.getLogger(MoveDataverseCommand.class.getName());
     final Dataverse moved;
     final Dataverse destination;
-    final Boolean force;
+    final boolean force;
 
-    public MoveDataverseCommand(DataverseRequest aRequest, Dataverse moved, Dataverse destination, Boolean force) {
+    public MoveDataverseCommand(DataverseRequest aRequest, Dataverse moved, Dataverse destination, boolean force) {
         super(aRequest, dv("moved", moved),
               dv("source", moved.getOwner()),
               dv("destination", destination));
@@ -116,7 +116,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
             List<Guestbook> movedGbs = moved.getGuestbooks();
             destinationGbs = destination.getGuestbooks();
             boolean inheritGuestbooksValue = !destination.isGuestbookRoot();
-            if (inheritGuestbooksValue && destination.getOwner() != null) {
+            if (inheritGuestbooksValue && !destination.isRoot()) {
                 destinationGbs.addAll(destination.getParentGuestbooks());
             }
             // include guestbooks in moved dataverse since they will also be there
@@ -130,7 +130,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
             logger.info("Checking featured dataverses...");
             for (DataverseFeaturedDataverse dfdv : ownerFeaturedDv) {
                 if (moved.equals(dfdv.getFeaturedDataverse())) {
-                    if (force == null || !force) {
+                    if (!force) {
                         removeFeatDv = true;
                         break;
                     }
@@ -162,7 +162,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
             inheritMbValue = !destination.isMetadataBlockRoot();
         }
 
-        List<DataverseLinkingDataverse> linkingDataverses = new ArrayList();
+        List<DataverseLinkingDataverse> linkingDataverses = new ArrayList<>();
 
         logger.info("Checking templates and metadata blocks");
         for (Dataverse dv : dataverseChildren) {
@@ -171,7 +171,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
             if (destinationTemplates != null) {
                 Template dvt = dv.getDefaultTemplate();
                 if (dvt != null && !destinationTemplates.contains(dvt)) {
-                    if (force == null || !force) {
+                    if (!force) {
                         removeTemplate = true;
                         break;
                     }
@@ -191,9 +191,9 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
                     MetadataBlock mb = iter.next();
                     // if the owner is null, it means that the owner is the root dataverse
                     // because technically only custom metadata blocks have owners
-                    Dataverse mbOwner = (mb.getOwner() != null) ? mb.getOwner() : ctxt.dataverses().findByAlias(":root");
+                    Dataverse mbOwner = (mb.getOwner() != null) ? mb.getOwner() : ctxt.dataverses().findRootDataverse();
                     if (!mbParentsToCheck.contains(mbOwner)) {
-                        if (force == null || !force) {
+                        if (!force) {
                             removeMetadataBlock = true;
                             break;
                         }
@@ -203,7 +203,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
                         metadataBlocksToKeep.add(mb);
                     }
                 }
-                if (force != null && force) {
+                if (force) {
                     dv.setMetadataBlocks(metadataBlocksToKeep);
                 }
             }
@@ -214,14 +214,14 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
             }
         }
 
-        List<DatasetLinkingDataverse> linkingDatasets = new ArrayList();
+        List<DatasetLinkingDataverse> linkingDatasets = new ArrayList<>();
         logger.info("Checking guestbooks...");
         for (Dataset ds : datasetChildren) {
             // if all the dataverse's datasets GUESTBOOKS are not 
             //contained in the new dataverse, then remove them
             Guestbook dsgb = ds.getGuestbook();
             if (dsgb != null && (destinationGbs == null || !destinationGbs.contains(dsgb))) {
-                if (force == null || !force) {
+                if (!force) {
                     removeGuestbook = true;
                     break;
                 }
@@ -240,7 +240,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
         for (DataverseLinkingDataverse dvld : linkingDataverses) {
             for (Dataverse owner : ownersToCheck) {
                 if ((dvld.getLinkingDataverse()).equals(owner)) {
-                    if (force == null || !force) {
+                    if (!force) {
                         removeLinkDv = true;
                         break;
                     }
@@ -257,7 +257,7 @@ public class MoveDataverseCommand extends AbstractVoidCommand {
         for (DatasetLinkingDataverse dsld : linkingDatasets) {
             for (Dataverse owner : ownersToCheck) {
                 if ((dsld.getLinkingDataverse()).equals(owner)) {
-                    if (force == null || !force) {
+                    if (!force) {
                         removeLinkDs = true;
                         break;
                     }

--- a/dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/engine/command/impl/MoveDataverseCommandTest.java
+++ b/dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/engine/command/impl/MoveDataverseCommandTest.java
@@ -305,14 +305,14 @@ public class MoveDataverseCommandTest {
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
 
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childB, childA, null));
+                new MoveDataverseCommand(aRequest, childB, childA, false));
 
         assertEquals(childA, childB.getOwner());
         assertEquals(Arrays.asList(root, childA), childB.getOwners());
 
         // move back
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childB, root, null));
+                new MoveDataverseCommand(aRequest, childB, root, false));
 
         assertEquals(root, childB.getOwner());
         assertEquals(Arrays.asList(root), childB.getOwners());
@@ -326,7 +326,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testInvalidMove");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childA, grandchildAA, null));
+                new MoveDataverseCommand(aRequest, childA, grandchildAA, false));
         fail();
     }
 
@@ -338,7 +338,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testNotSuperUser");
         DataverseRequest aRequest = new DataverseRequest(nobody, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childB, childA, null));
+                new MoveDataverseCommand(aRequest, childB, childA, false));
         fail();
     }
 
@@ -347,7 +347,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testMoveIntoSelf");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childB, childB, null));
+                new MoveDataverseCommand(aRequest, childB, childB, false));
         fail();
     }
 
@@ -356,7 +356,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testMoveIntoParent");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, grandchildAA, childA, null));
+                new MoveDataverseCommand(aRequest, grandchildAA, childA, false));
         fail();
     }
 
@@ -365,12 +365,12 @@ public class MoveDataverseCommandTest {
         System.out.println("testKeepGuestbook");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childC, childB, null));
+                new MoveDataverseCommand(aRequest, childC, childB, false));
         assertNotNull(datasetC.getGuestbook());
 
         // move back
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childC, root, null));
+                new MoveDataverseCommand(aRequest, childC, root, false));
         assertEquals(root, childC.getOwner());
     }
 
@@ -379,7 +379,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testRemoveGuestbookWithoutForce");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, grandchildCC, root, null));
+                new MoveDataverseCommand(aRequest, grandchildCC, root, false));
         fail();
     }
 
@@ -402,12 +402,12 @@ public class MoveDataverseCommandTest {
         System.out.println("testKeepTemplate");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childD, childB, null));
+                new MoveDataverseCommand(aRequest, childD, childB, false));
         assertNotNull(grandchildDD.getDefaultTemplate());
 
         // move back
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childD, root, null));
+                new MoveDataverseCommand(aRequest, childD, root, false));
         assertEquals(root, childD.getOwner());
 
     }
@@ -417,7 +417,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testRemoveTemplateWithoutForce");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, grandchildDD, root, null));
+                new MoveDataverseCommand(aRequest, grandchildDD, root, false));
         fail();
     }
 
@@ -440,12 +440,12 @@ public class MoveDataverseCommandTest {
         System.out.println("testKeepMetadataBlock");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childE, childB, null));
+                new MoveDataverseCommand(aRequest, childE, childB, false));
         assertEquals(Arrays.asList(mbB), childE.getRootMetadataBlocks());
 
         // move back
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, childE, root, null));
+                new MoveDataverseCommand(aRequest, childE, root, false));
         assertEquals(root, childE.getOwner());
     }
 
@@ -454,7 +454,7 @@ public class MoveDataverseCommandTest {
         System.out.println("testRemoveMetadataBlockWithoutForce");
         DataverseRequest aRequest = new DataverseRequest(auth, httpRequest);
         testEngine.submit(
-                new MoveDataverseCommand(aRequest, grandchildEE, root, null));
+                new MoveDataverseCommand(aRequest, grandchildEE, root, false));
         fail();
     }
 

